source/extensions/filters/http/fault/fault_filter.cc

```diff
@@ -318,8 +357,101 @@ void FaultFilter::resetTimerState() {
   }
 }
 
-void FaultFilter::setDecoderFilterCallbacks(Http::StreamDecoderFilterCallbacks& callbacks) {
-  callbacks_ = &callbacks;
+Http::FilterDataStatus FaultFilter::encodeData(Buffer::Instance& data, bool end_stream) {
+  if (response_limiter_ != nullptr) {
+    response_limiter_->writeData(data, end_stream);
+    return Http::FilterDataStatus::StopIterationNoBuffer;
```

Comment:
what does no buffer do, when upstream is sending constant stream of data for progressive download?

Reply:
It doesn't buffer any data. The filter handles everything. Can you describe your concern in more detail?

Reply:
I was just curious